docs/proposals/config_data.md

```diff
@@ -0,0 +1,298 @@
+                fieldPath: component_metadata
+```
+
+Now, we create the replication controller:
+```bash
+kubectl create -f example-replication-controller.yaml
+```
+
+New pods created by the ExampleReplicationController will use the ``pod-config-volume`` objects when starting/replicating pods.
+
+``pod-config-volume`` object will be watched for any modifications and the mounted volumes will be updated if there are any modifications.
+
+#### Consuming ConfigData as environment variables
+
+```json
+{
+    "apiVersion": "extensions/v1beta1",
+    "metadata": {
+        "name": "pod-config-env",
+    },
+    "kind": "ConfigData",
+    "data": {
+        "api_server": "10.240.13.14:8998",
+        "api_version": "extensions/v1beta1",
+        "db_host": "10.240.55.3",
+        "db_port": "3306"
+    }
+}
+```
+``pod-config-env`` is intended to be consumed as environment variable key/value pairs.
+
+We create our ``ConfigData`` object.
+```bash
+kubectl create -f pod-configuration-env.json
+```
+
+ExampleReplicationController starts 2 replicas that runs the `busybox` container.
+In this use case, the ``ConfigData`` object will be consumed as 4 environment variables.
+```yaml
+apiVersion: extensions/v1beta1
+kind: ReplicationController
+metadata:
+  name: ExampleReplicationController
+spec:
+  replicas: 2
+  selector:
+    app: busybox
+  template:
+    metadata:
+      labels:
+        app: busybox
+    spec:
+      containers:
+      - name: busybox
+        image: gcr.io/google_containers/busybox
+        ports:
+        - containerPort: 80
+        env:
+        - name: API_SERVER
+          valueFrom:
+            configDataRef:
+              from: pod-config-env
+              key: api_servers
+        - name: API_VERSION
+          valueFrom:
+            configDataRef:
+              from: pod-config-env
+              key: api_version
+        - name: DB_HOST
+          valueFrom:
+            configDataRef:
+              from: pod-config-env
+              key: postgresql_host
+        - name: DB_PORT
+          valueFrom:
+            configDataRef:
+              from: pod-config-env
+              key: postgresql_port
+```
+
+Now, we create the replication controller:
+```bash
+kubectl create -f example-replication-controller.yaml
+```
+
+New pods created by the ``ExampleReplicationController`` will use the ``pod-config-env`` objects when starting/replicating pods.
+
+``pod-config-env`` object will be watched for any modifications and the currently consumed environment variables will be updated.
+
+### Future Improvements
+
+* Additional specification parameters can be added to the API for describing the expected behavior after
+a modification to the consumed ``ConfigData`` object (e.g. 'on-restart: update').
+
+### Considerations
+
+* Multiple ``ConfigData`` objects can be supplied to the component for overriding or different consuming purposes (one for environment variables the other to be mounted as a volume)
```

Comment:
I don't understand this section -- let's strike it unless it makes a really important point.